app/src/main/java/org/mozilla/fenix/addons/InstalledAddonDetailsFragment.kt

```diff
@@ -211,6 +212,28 @@ class InstalledAddonDetailsFragment : Fragment() {
         }
     }
 
+    private fun bindAllowInPrivateBrowsingSwitch(view: View) {
+        val switch = view.allow_in_private_browsing_switch
+        switch.isChecked = addon.isAllowedInPrivateBrowsing()
+        switch.setOnCheckedChangeListener { v, isChecked ->
+            val addonManager = v.context.components.addonManager
+            switch.isClickable = false
+            view.remove_add_on.isEnabled = false
+            addonManager.setAddonAllowedInPrivateBrowsing(
+                addon,
+                isChecked,
+                onSuccess = {
+                    runIfFragmentIsAttached {
+                        this.addon = it
+                        switch.isClickable = true
+                        switch.isChecked = isChecked
```

Comment:
Is this going to re-triger `setOnCheckedChangeListener`? Why do we need to change its value again?

Reply:
We do the same thing `Switch.setState()` for enable/disable.